export/export.go

```diff
@@ -0,0 +1,438 @@
+type ResourcesForImport struct {
+	Pods            []v1.PodApplyConfiguration                        `json:"pods"`
+	Nodes           []v1.NodeApplyConfiguration                       `json:"nodes"`
+	Pvs             []v1.PersistentVolumeApplyConfiguration           `json:"pvs"`
+	Pvcs            []v1.PersistentVolumeClaimApplyConfiguration      `json:"pvcs"`
+	StorageClasses  []confstoragev1.StorageClassApplyConfiguration    `json:"storageClasses"`
+	PriorityClasses []schedulingcfgv1.PriorityClassApplyConfiguration `json:"priorityClasses"`
+	SchedulerConfig *v1beta2config.KubeSchedulerConfiguration         `json:"schedulerConfig"`
+}
+
+type PodService interface {
+	List(ctx context.Context) (*corev1.PodList, error)
+	Apply(ctx context.Context, pod *v1.PodApplyConfiguration) (*corev1.Pod, error)
+}
+
+type NodeService interface {
+	List(ctx context.Context) (*corev1.NodeList, error)
+	Apply(ctx context.Context, nac *v1.NodeApplyConfiguration) (*corev1.Node, error)
+}
+
+type PersistentVolumeService interface {
+	List(ctx context.Context) (*corev1.PersistentVolumeList, error)
+	Apply(ctx context.Context, persistentVolume *v1.PersistentVolumeApplyConfiguration) (*corev1.PersistentVolume, error)
+}
+
+type PersistentVolumeClaimService interface {
+	Get(ctx context.Context, name string) (*corev1.PersistentVolumeClaim, error)
+	List(ctx context.Context) (*corev1.PersistentVolumeClaimList, error)
+	Apply(ctx context.Context, persistentVolumeClaime *v1.PersistentVolumeClaimApplyConfiguration) (*corev1.PersistentVolumeClaim, error)
+}
+
+type StorageClassService interface {
+	List(ctx context.Context) (*storagev1.StorageClassList, error)
+	Apply(ctx context.Context, storageClass *confstoragev1.StorageClassApplyConfiguration) (*storagev1.StorageClass, error)
+}
+
+type PriorityClassService interface {
+	List(ctx context.Context) (*schedulingv1.PriorityClassList, error)
+	Apply(ctx context.Context, priorityClass *schedulingcfgv1.PriorityClassApplyConfiguration) (*schedulingv1.PriorityClass, error)
+}
+
+type SchedulerService interface {
+	GetSchedulerConfig() *v1beta2config.KubeSchedulerConfiguration
+	RestartScheduler(cfg *v1beta2config.KubeSchedulerConfiguration) error
+}
+
+func NewExportService(client clientset.Interface, pods PodService, nodes NodeService, pvs PersistentVolumeService, pvcs PersistentVolumeClaimService, storageClasss StorageClassService, priorityClasss PriorityClassService, schedulers SchedulerService) *Service {
+	return &Service{
+		client:               client,
+		podService:           pods,
+		nodeService:          nodes,
+		pvService:            pvs,
+		pvcService:           pvcs,
+		storageClassService:  storageClasss,
+		priorityclassService: priorityClasss,
+		schedulerService:     schedulers,
+	}
+}
+
+type ErrorGroupWithSemaphore struct {
+	g   *errgroup.Group
+	sem *semaphore.Weighted
+}
+
+func NewErrorGroupWithSemaphore(ctx context.Context) ErrorGroupWithSemaphore {
```

Comment:
This `ErrorGroupWithSemaphore` is a nice idea 👍  
Could you move this to the util package so that we can use it from other packages?

https://github.com/kubernetes-sigs/kube-scheduler-simulator/tree/master/util

Reply:
And one nits: rename this from `ErrorGroupWithSemaphore` to `ErrGroupWithSemaphore`